src/Language/Fortran/Repr/Value/Machine.hs

```diff
@@ -18,3 +20,14 @@ data FValue = MkFScalarValue FScalarValue
 fValueType :: FValue -> FType
 fValueType = \case
   MkFScalarValue a -> MkFScalarType $ fScalarValueType a
+
+fromConstInt :: FValue -> Maybe Integer
+fromConstInt (MkFScalarValue (FSVInt a)) = Just $ withFInt a
+fromConstInt _          = Nothing
+
+fromConstReal :: FValue -> Maybe Double
+fromConstReal (MkFScalarValue (FSVReal (FReal4 a))) = Just $ floatToDouble a
+  where
+    floatToDouble :: Float -> Double
+    floatToDouble = realToFrac
+fromConstReal (MkFScalarValue (FSVReal (FReal8 a))) = Just $ a
```

Comment:
`fromConstReal` is currently partial. Probably want a fallback case like `fromConstReal _ = Nothing`

Reply:
Thanks. Fixed in 5bd841f